### ISSUE TITLE
PMIDs missing from citation datasets

### ISSUE THREAD

**Issue by kmartinez834** [glygener/glygen-issues#84]
Datasets with PMIDs that are missing in the corresponding citations dataset:

human_proteoform_phosphorylation_sites_iptmnet.csv
92065884 - PMID is not available
91378515 - PMID is not available
81086167 - PMID is not available
93076812 - PMID is not available
91085562 - PMID is not available
95059042 - PMID is not available
90059885 - PMID is not available
91184134 - PMID is not available
169175071 - PMID is not available
106488285 - PMID is not available
30097426 - PMID is not available
24795023 - PMID file missing required fields
29964354 - PMID file missing required fields
28958491 - duplicate of PMID 26844273
31101629 - duplicate of PMID 30944118
31088862 - duplicate of PMID 31267715

mouse_proteoform_phosphorylation_sites_iptmnet.csv
31224398 - PMID is not available

human_protein_biomarkers.csv
21720365 - PMID file missing required fields

human_proteoform_glycosylation_sites_glyconnect.csv
1687959 - PMID file missing required fields

mouse_proteoform_glycosylation_sites_o_glcnac_mcw.csv
34004831 - duplicate of PMID 32165492

human_protein_mutation_somatic.csv
22810696 - PMID file missing required fields
18772890 - PMID file missing required fields
24292195 - PMID file missing required fields
21720365 - PMID file missing required fields
10786688 - No such file or directory

rat_protein_function_refseq.csv
_1155 PMIDs missing, some examples:_
33686740 - No such file or directory
31721324 - No such file or directory
35181828 - No such file or directory
35257794 - No such file or directory
35183034 - No such file or directory
34920725 - No such file or directory
19168440 - No such file or directory
35070875 - No such file or directory

human_protein_function_refseq.csv
_11,573 PMIDs missing, some examples:_
35955722 - No such file or directory
35886055 - No such file or directory
35290903 - No such file or directory
35768785 - No such file or directory
35848475 - No such file or directory

human_protein_reactions_reactome.csv
_1993 PMIDs missing, some examples:_
 20504255 - No such file or directory
 30716294 - No such file or directory
 31520644 - Valid PMID file
 9605565 - No such file or directory
20173735 - No such file or directory

downloads/glytoucan/current/export/pubs.tsv
9398638 - PMID is not available
77532821 - PMID is not available

glycan_motif.csv
_29 PMIDs missing
PMIDs are separated by ";" Some examples:_
7287743; 19756298
2410529; 19756298
1567687; 6653876; 19756298
281705; 6165896; 19756298

mouse_protein_function_refseq.csv
_4808 PMIDs missing, some examples:_
35596790 - No such file or directory
35802383 - No such file or directory
35705936 - No such file or directory
35269933 - No such file or directory
36005758 - No such file or directory

human_protein_mutation_literature.csv
_4386 PMIDs missing, some examples:_
27515171 - No such file or directory
25245010 - No such file or directory
15754732 - No such file or directory
23526598 - No such file or directory
28173668 - No such file or directory

**Comment by jeet-vora** [comment]
Files like 
rat_protein_matrixdb.csv contain nonPMIDs in the evidence field

'''"P55067-1","GAG_1","","CHEBI:28304","psi-mi:MI:0499(unspecified role)","psi-mi:MI:0499(unspecified role)","imex:IM-14084","","Glycosaminoglycan","MatrixDB","matrixdb:GAG_1","protein_xref_pubmed","IM-14084","glycan_xref_matrixdb","GAG_1"'''

'''"P55067-1","GAG_9","","CHEBI:16336","psi-mi:MI:0499(unspecified role)","psi-mi:MI:0499(unspecified role)","imex:IM-14084",","Glycosaminoglycan","MatrixDB","matrixdb:GAG_9","glycan_xref_matrixdb","GAG_9","glycan_xref_matrixdb","GAG_9"'''

**Comment by rykahsay** [comment]
Please check again, and also try to have corresponding logfile under logs/  --> it will try to explain why that PMID is missing.

BTW, the source of the PMIDs is the file under the unreviewed folder, so you should not compare agains PMID containing downloaded/input files such as:

downloads/glytoucan/current/export/pubs.tsv
9398638
77532821

**Comment by kmartinez834** [comment]
@rykahsay -->

1. "BTW, the source of the PMIDs is the file under the unreviewed folder, so you should not compare agains PMID containing downloaded/input files such as: downloads/glytoucan/current/export/pubs.tsv"

The file glycan_citations_glytoucan.csv does not have an input in unreviewed folder w/ PMIDs (see code block below).  See issue #80 with other source files that the check-citation-datasets.py script was not able to find.
```
$ cat glycan_citations_glytoucan.fu.log 
/data/projects/glygen/generated/datasets/unreviewed/glycan_masterlist.csv
/data/shared/glyds/downloads/glytoucan/gtc_12_20_2022/export/pubs.tsv
```

3. I am getting the following error when I try to re-run the script:
```
$ python3 check-citation-datasets.py 
diff,src_file,idcount_in_src_file,citations_file,idcount_in_citations_file
17,human_proteoform_phosphorylation_sites_iptmnet.csv,22130,human_proteoform_citations_phosphorylation_sites_iptmnet.csv,22113
Traceback (most recent call last):
  File "check-citation-datasets.py", line 94, in <module>
    main()
  File "check-citation-datasets.py", line 83, in main
    with open(log_file, "w") as FW:
PermissionError: [Errno 13] Permission denied: 'logs/human_proteoform_citations_phosphorylation_sites_iptmnet.missing.log'
```

**Comment by kmartinez834** [comment]
@rykahsay -->

I updated the PMID list above to indicate why they were excluded.  Most are unavailable, missing required fields, or haven't been downloaded yet.  

1. I found one PMID (31520644) from `human_protein_reactions_reactome.csv` that appears to be valid, but is not present in `human_protein_citations_reactome.csv`.  It's not listed in the log file either.   Any suggestions for why?

2. Some of the Medline files indicate that the PMID is a duplicate:  
```
$ cat /data/projects/glygen/downloads/ncbi/medline/pmid.31088862.txt
id: 31088862 Error occurred: PMID 31088862 is a duplicate of PMID 31267715
```
   - I started a mapping file with "pmid" and "duplicate" fields in `misc/dup_pmid_mapping.csv`
